annot/src/main/java/com/predic8/membrane/annot/yaml/BeanRegistryImplementation.java

```diff
@@ -35,27 +35,78 @@ public class BeanRegistryImplementation implements BeanRegistry {
     /**
      * TODO Rename give meaningful name
      */
-    private final ConcurrentHashMap<String, Object> uuidMap = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<String, Object> uuidMap = new ConcurrentHashMap<>(); // Order is here not critical
 
     private final BlockingQueue<ChangeEvent> changeEvents = new LinkedBlockingDeque<>();
 
     // uid -> bean definition
-    private final Map<String, BeanDefinition> bds = new ConcurrentHashMap<>();
-    private final Set<String> uidsToActivate = ConcurrentHashMap.newKeySet();
+    private final Map<String, BeanDefinition> bds = new ConcurrentHashMap<>(); // Order is not critical. Order is determined by uidsToActivate
+    private final Set<String> uidsToActivate = new LinkedHashSet<>(); // Provides order
 
     public BeanRegistryImplementation(BeanCacheObserver observer, Grammar grammar) {
         this.observer = observer;
         this.grammar = grammar;
     }
 
+    /**
+     * Registers a set of {@link BeanDefinition}s originating from static configuration
+     * (for example YAML files on startup).
+     *
+     * <p>Thread–safety model:</p>
+     * <ul>
+     *   <li>This method is expected to be called from a single thread.</li>
+     *   <li>It publishes initialization events into the internal change queue but does not
+     *       perform any activation work itself.</li>
+     *   <li>Callers must ensure that no other thread invokes this method
+     *       concurrently.</li>
+     *   <li>Dynamic updates (e.g. from Kubernetes watchers) must use {@link #handle(WatchAction, JsonNode)}
+     *       instead; these calls are thread-safe because they only enqueue events.</li>
+     * </ul>
+     *
+     * <p>Lifecycle expectations:</p>
+     * <ul>
+     *   <li>After publishing all static configuration events, this method signals that
+     *       the configuration stream has finished by inserting a {@code StaticConfigurationLoaded}
+     *       marker event.</li>
+     *   <li>The caller must invoke {@link #start()} exactly once after registration to process all
+     *       pending events and activate the beans.</li>
+     * </ul>
+     *
+     * @param bds the list of static {@link BeanDefinition}s to register
+     */
```

Comment:
The comment is very similar but different (❗) from the comment on the interface.

Please not not add auto-generated inconsistent comments.